sapirbitan/assignment1/ex2.cpp

```diff
@@ -0,0 +1,46 @@
+#include <iostream>
+#include <string>
+using namespace std;
+
+struct some_struct {
+	int num;
+	some_struct* s=NULL;
+};
+/* I am assuming each element in the linked list contains a pointer to the next element in the list,
+also, I am assuming my function gets the "head" of the linked list and 'k'	*/
+
+some_struct find_kth_element(some_struct first_element, int k) {
```

Comment:
As I mentioned in a comment on Q1, the function signature should be:

`some_struct* find_kth_element(const some_struct& first_element, int k)`

Also, you can just rename 'first_element' to 'head', it's more clearer.